EdFi.Ods.Utilities.Migration/Providers/MigrationConfigurationProvider.cs

```diff
@@ -62,21 +62,21 @@ public MigrationConfigurationProvider(IOdsMigrationManagerResolver odsMigrationM
                 ? _odsMigrationManagerResolver.GetLatestSupportedUpgradeVersion(upgradeVersionConfiguration.VersionBeforeUpgrade)
                 : GetRequestedUpgradeVersionFromUserProvidedString(optionalRequestedUpgradeVersionOverride);
 
-            RaiseErrorIfUpgradePathNotSupported(upgradeVersionConfiguration);
+            RaiseErrorIfUpgradePathNotSupported(upgradeVersionConfiguration, options);
 
             upgradeVersionConfiguration.FeaturesBeforeUpgrade =
                 autoDetectedVersion?.ExistingFeatures?.ToList() ?? new List<EdFiOdsFeature>();
 
             return upgradeVersionConfiguration;
         }
 
-        private void RaiseErrorIfUpgradePathNotSupported(UpgradeVersionConfiguration upgradeVersionConfiguration)
+        private void RaiseErrorIfUpgradePathNotSupported(UpgradeVersionConfiguration upgradeVersionConfiguration, Options options)
         {
-            if (!_odsMigrationManagerResolver.VersionCanBeUpgraded(upgradeVersionConfiguration.VersionBeforeUpgrade))
+            if (!_odsMigrationManagerResolver.VersionCanBeUpgraded(upgradeVersionConfiguration.VersionBeforeUpgrade, options.Engine))
             {
                 throw new ArgumentException($"Upgrade for ODS Version {upgradeVersionConfiguration.VersionBeforeUpgrade} is not currently supported" +
                                             $"{Environment.NewLine}The following versions can be upgraded:  " +
-                                            $"{ _odsMigrationManagerResolver.GetAllUpgradableVersions().ToBulletedList() }");
+                                            $"{ _odsMigrationManagerResolver.GetAllUpgradableVersions(options.Engine).ToBulletedList() }");
```

Comment:
Agreed with Travis' comment on this, let's update the interface and implementation to never default the engine, force consumers to choose.

Reply:
Removed the defaults and ran the tests locally. Looks good.